internal/gke/metrics_test.go

```diff
@@ -0,0 +1,194 @@
+	return nil
+}
+
+func (m *metricsApiClientMock) Flags(ctx context.Context) (v1.FlagsResult, error) {
+	return nil, nil
+}
+
+func (m *metricsApiClientMock) LabelNames(ctx context.Context, matches []string, startTime, endTime time.Time) ([]string, v1.Warnings, error) {
+	return nil, nil, nil
+}
+
+func (m *metricsApiClientMock) LabelValues(ctx context.Context, label string, matches []string, startTime, endTime time.Time) (model.LabelValues, v1.Warnings, error) {
+	return nil, nil, nil
+}
+
+func (m *metricsApiClientMock) QueryRange(ctx context.Context, query string, r v1.Range, opts ...v1.Option) (model.Value, v1.Warnings, error) {
+	return nil, nil, nil
+}
+
+func (m *metricsApiClientMock) QueryExemplars(ctx context.Context, query string, startTime, endTime time.Time) ([]v1.ExemplarQueryResult, error) {
+	return nil, nil
+}
+
+func (m *metricsApiClientMock) Buildinfo(ctx context.Context) (v1.BuildinfoResult, error) {
+	return v1.BuildinfoResult{}, nil
+}
+
+func (m *metricsApiClientMock) Runtimeinfo(ctx context.Context) (v1.RuntimeinfoResult, error) {
+	return v1.RuntimeinfoResult{}, nil
+}
+
+func (m *metricsApiClientMock) Series(ctx context.Context, matches []string, startTime, endTime time.Time) ([]model.LabelSet, v1.Warnings, error) {
+	return nil, nil, nil
+}
+
+func (m *metricsApiClientMock) Snapshot(ctx context.Context, skipHead bool) (v1.SnapshotResult, error) {
+	return v1.SnapshotResult{}, nil
+}
+
+func (m *metricsApiClientMock) Rules(ctx context.Context) (v1.RulesResult, error) {
+	return v1.RulesResult{}, nil
+}
+
+func (m *metricsApiClientMock) Targets(ctx context.Context) (v1.TargetsResult, error) {
+	return v1.TargetsResult{}, nil
+}
+
+func (m *metricsApiClientMock) TargetsMetadata(ctx context.Context, matchTarget, metric, limit string) ([]v1.MetricMetadata, error) {
+	return nil, nil
+}
+
+func (m *metricsApiClientMock) Metadata(ctx context.Context, metric, limit string) (map[string][]v1.Metadata, error) {
+	return nil, nil
+}
+
+func (m *metricsApiClientMock) TSDB(ctx context.Context) (v1.TSDBResult, error) {
+	return v1.TSDBResult{}, nil
+}
+
+func (m *metricsApiClientMock) WalReplay(ctx context.Context) (v1.WalReplayStatus, error) {
+	return v1.WalReplayStatus{}, nil
+}
+
+func TestNewMetricClient(t *testing.T) {
+	ctx := context.TODO()
+	cli, err := NewMetricClient(ctx, "test-project", "fake-token")
+	if err != nil {
+		t.Fatalf("err is not nil; want nil; err = %s", err)
+	}
+	realCli, ok := cli.(*metricsClient)
+	if !ok {
+		t.Fatalf("cli is not *gke.metricsClient")
+	}
+	if realCli.ctx != ctx {
+		t.Errorf("context is %v; want %v", realCli.ctx, ctx)
+	}
+	if realCli.client == nil {
+		t.Errorf("client is nil; want api.Client")
+	}
+}
+
+func TestGetMetric(t *testing.T) {
+
+	metricName := "test-metric"
+	metricValue := 22
+
+	v1ApiMock := &metricsApiClientMock{
+		QueryFn: func(ctx context.Context, query string, ts time.Time, opts ...v1.Option) (model.Value, v1.Warnings, error) {
+
```

Comment:
You can also test here that received `query` string matches inputs passed to `GetMetric` func